api/auth.proto

```diff
@@ -44,3 +46,78 @@ message AuthAction {
   // List of rules
   repeated Rule rule = 2;
 }
+
+service Authorization {
+	rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+
+	// Address of the downstream source of the request.
+	Address source = 1;
+
+	// Address of the upstream destination of the request.
+	Address destination = 2; 
+
+	// If the request is an HTTP request, the parameters
+	// of the HTTP request.
+	HTTPRequest http = 3;
+
+	// If the request came over a secure TLS connection, information about the connection and 
+	// downstream client.
+	TLSConnection tls = 4;
+}
+
+message HTTPRequest {
+
+	// HTTP Version of the request, e.g. "1.1"
+	string version = 1;
```

Comment:
This is related to the ongoing conversation about headers in https://github.com/envoyproxy/data-plane-api/pull/237. Can we wait until that is resolved and do the same thing here? If we end up with explicit headers, we should consider sharing a message.

Reply:
I wonder if there can be greater unification here; is there a common message that is useful to both external auth checking and access logging?

Reply:
Sounds good.  Should we just reuse `HTTPRequestProperties` or does that contain information generated internal to the access log filter?  If so we could share a message that is the intersection with this.